src/main/java/com/example/bookstoreapp/model/User.java

```diff
@@ -0,0 +1,44 @@
+package com.example.bookstoreapp.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.persistence.UniqueConstraint;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted = false")
+@Table(name = "users", uniqueConstraints = {@UniqueConstraint(columnNames = {"email"})})
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true)
```

Comment:
> 

Could not find @Unique annotation it might be
@UniqueElements - from hibernate or 
@UniqueConstraint - from jakarta.persistence
but it seams for me it would be much simpler to use just @Column(nullable = false, unique = true)